client/app/collections/upload_queue.coffee

```diff
@@ -93,60 +103,87 @@ module.exports = class UploadQueue extends Backbone.Collection
         @trigger 'upload-complete'
 
 
-    uploadWorker: (model, callback) =>
-        if model.existing or model.error
-            setTimeout callback, 10
+    # When an upload completes or fails, we keep counts.
+    onSyncError: (model) =>
+        path = model.get('path') + '/'
+        @uploadingPaths[path]--
+        @loaded++
+
+
+    # Process each element from the queue
+    uploadWorker: (model, next) =>
+        # Skip if there is an error.
+        if model.error
+            setTimeout next, 10
+
+        # If there is a conflict, the queue waits for the user to
+        # make a decision.
+        else if model.isConflict()
+            model.processOverwrite = @_decideOn.bind @, model, next
```

Comment:
I'm feeling uneasy with explaining native function: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind. I'll try to do something though.